<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>StakingPage</class>
 <widget class="QWidget" name="StakingPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Staking - Shahcoin Core</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>12</number>
   </property>
   <property name="leftMargin">
    <number>12</number>
   </property>
   <property name="topMargin">
    <number>12</number>
   </property>
   <property name="rightMargin">
    <number>12</number>
   </property>
   <property name="bottomMargin">
    <number>12</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox_StakingInfo">
     <property name="title">
      <string>üí∞ Staking Information</string>
     </property>
     <property name="styleSheet">
      <string>QGroupBox {
    font-weight: bold;
    font-size: 14px;
    color: #2E8B57;
}</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <property name="spacing">
       <number>8</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_TotalBalance">
        <property name="text">
         <string>Total Balance:</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold; font-size: 13px;</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_TotalBalanceValue">
        <property name="text">
         <string>0.00000000 SHAH</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold; color: #2E8B57; font-size: 13px;</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_StakedBalance">
        <property name="text">
         <string>Staked Balance:</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold; font-size: 13px;</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="label_StakedBalanceValue">
        <property name="text">
         <string>0.00000000 SHAH</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold; color: #4169E1; font-size: 13px;</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_StakeableBalance">
        <property name="text">
         <string>Available for Staking:</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold; font-size: 13px;</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="label_StakeableBalanceValue">
        <property name="text">
         <string>0.00000000 SHAH</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold; color: #FF6347; font-size: 13px;</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_EstimatedRewards">
        <property name="text">
         <string>Estimated Annual Rewards:</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold; font-size: 13px;</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="label_EstimatedRewardsValue">
        <property name="text">
         <string>0.00000000 SHAH</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold; color: #32CD32; font-size: 13px;</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_StakeAge">
        <property name="text">
         <string>Minimum Stake Age:</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold; font-size: 13px;</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLabel" name="label_StakeAgeValue">
        <property name="text">
         <string>12 hours</string>
        </property>
        <property name="styleSheet">
         <string>font-size: 13px;</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_MinStake">
        <property name="text">
         <string>Minimum Stake Amount:</string>
        </property>
        <property name="styleSheet">
         <string>font-weight: bold; font-size: 13px;</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLabel" name="label_MinStakeValue">
        <property name="text">
         <string>333.00000000 SHAH</string>
        </property>
        <property name="styleSheet">
         <string>font-size: 13px;</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_StakingActions">
     <property name="title">
      <string>üîí Staking Actions</string>
     </property>
     <property name="styleSheet">
      <string>QGroupBox {
    font-weight: bold;
    font-size: 14px;
    color: #4169E1;
}</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="spacing">
       <number>12</number>
      </property>
      <item>
       <widget class="QPushButton" name="pushButton_Stake">
        <property name="minimumSize">
         <size>
          <width>120</width>
          <height>40</height>
         </size>
        </property>
        <property name="text">
         <string>üí∞ Start Staking</string>
        </property>
        <property name="toolTip">
         <string>Start staking with your available balance (minimum 333 SHAH required)</string>
        </property>
        <property name="styleSheet">
         <string>QPushButton {
    background-color: #4CAF50;
    border: none;
    color: white;
    padding: 8px 16px;
    text-align: center;
    font-size: 14px;
    margin: 4px 2px;
    border-radius: 6px;
    font-weight: bold;
}
QPushButton:hover {
    background-color: #45a049;
    transform: translateY(-1px);
    box-shadow: 0 2px 4px rgba(0,0,0,0.2);
}
QPushButton:pressed {
    background-color: #3d8b40;
    transform: translateY(0px);
}
QPushButton:disabled {
    background-color: #cccccc;
    color: #666666;
    transform: none;
    box-shadow: none;
}</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_StopStaking">
        <property name="minimumSize">
         <size>
          <width>120</width>
          <height>40</height>
         </size>
        </property>
        <property name="text">
         <string>üîì Stop Staking</string>
        </property>
        <property name="toolTip">
         <string>Stop staking and unlock your staked coins</string>
        </property>
        <property name="styleSheet">
         <string>QPushButton {
    background-color: #f44336;
    border: none;
    color: white;
    padding: 8px 16px;
    text-align: center;
    font-size: 14px;
    margin: 4px 2px;
    border-radius: 6px;
    font-weight: bold;
}
QPushButton:hover {
    background-color: #da190b;
    transform: translateY(-1px);
    box-shadow: 0 2px 4px rgba(0,0,0,0.2);
}
QPushButton:pressed {
    background-color: #b71c1c;
    transform: translateY(0px);
}
QPushButton:disabled {
    background-color: #cccccc;
    color: #666666;
    transform: none;
    box-shadow: none;
}</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_ClaimRewards">
        <property name="minimumSize">
         <size>
          <width>120</width>
          <height>40</height>
         </size>
        </property>
        <property name="text">
         <string>üéÅ Claim Rewards</string>
        </property>
        <property name="toolTip">
         <string>Claim your staking rewards</string>
        </property>
        <property name="styleSheet">
         <string>QPushButton {
    background-color: #FF9800;
    border: none;
    color: white;
    padding: 8px 16px;
    text-align: center;
    font-size: 14px;
    margin: 4px 2px;
    border-radius: 6px;
    font-weight: bold;
}
QPushButton:hover {
    background-color: #F57C00;
    transform: translateY(-1px);
    box-shadow: 0 2px 4px rgba(0,0,0,0.2);
}
QPushButton:pressed {
    background-color: #E65100;
    transform: translateY(0px);
}
QPushButton:disabled {
    background-color: #cccccc;
    color: #666666;
    transform: none;
    box-shadow: none;
}</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_Refresh">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>40</height>
         </size>
        </property>
        <property name="text">
         <string>‚ôªÔ∏è Refresh</string>
        </property>
        <property name="toolTip">
         <string>Refresh staking information</string>
        </property>
        <property name="styleSheet">
         <string>QPushButton {
    background-color: #2196F3;
    border: none;
    color: white;
    padding: 8px 16px;
    text-align: center;
    font-size: 14px;
    margin: 4px 2px;
    border-radius: 6px;
    font-weight: bold;
}
QPushButton:hover {
    background-color: #0b7dda;
    transform: translateY(-1px);
    box-shadow: 0 2px 4px rgba(0,0,0,0.2);
}
QPushButton:pressed {
    background-color: #0d47a1;
    transform: translateY(0px);
}
QPushButton:disabled {
    background-color: #cccccc;
    color: #666666;
    transform: none;
    box-shadow: none;
}</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_ActiveStakes">
     <property name="title">
      <string>üìä Active Stakes</string>
     </property>
     <property name="styleSheet">
      <string>QGroupBox {
    font-weight: bold;
    font-size: 14px;
    color: #2E8B57;
}</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="spacing">
       <number>8</number>
      </property>
      <item>
       <widget class="QTableWidget" name="tableWidget_ActiveStakes">
        <property name="alternatingRowColors">
         <bool>true</bool>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <property name="sortingEnabled">
         <bool>true</bool>
        </property>
        <property name="styleSheet">
         <string>QTableWidget {
    gridline-color: #d0d0d0;
    background-color: white;
    alternate-background-color: #f8f8f8;
}
QTableWidget::item {
    padding: 4px;
    border-bottom: 1px solid #e0e0e0;
}
QTableWidget::item:selected {
    background-color: #e3f2fd;
    color: black;
}
QHeaderView::section {
    background-color: #f5f5f5;
    padding: 6px;
    border: 1px solid #d0d0d0;
    font-weight: bold;
}</string>
        </property>
        <column>
         <property name="text">
          <string>Stake ID</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Amount</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Age</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Rewards Earned</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Status</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_StakingHistory">
     <property name="title">
      <string>üìà Staking History</string>
     </property>
     <property name="styleSheet">
      <string>QGroupBox {
    font-weight: bold;
    font-size: 14px;
    color: #4169E1;
}</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <property name="spacing">
       <number>8</number>
      </property>
      <item>
       <widget class="QTableWidget" name="tableWidget_StakingHistory">
        <property name="alternatingRowColors">
         <bool>true</bool>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <property name="sortingEnabled">
         <bool>true</bool>
        </property>
        <property name="styleSheet">
         <string>QTableWidget {
    gridline-color: #d0d0d0;
    background-color: white;
    alternate-background-color: #f8f8f8;
}
QTableWidget::item {
    padding: 4px;
    border-bottom: 1px solid #e0e0e0;
}
QTableWidget::item:selected {
    background-color: #e3f2fd;
    color: black;
}
QHeaderView::section {
    background-color: #f5f5f5;
    padding: 6px;
    border: 1px solid #d0d0d0;
    font-weight: bold;
}</string>
        </property>
        <column>
         <property name="text">
          <string>Date</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Type</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Amount</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Reward</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Block Height</string>
         </property>
        </column>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
